cmd/collectors/rest/templating.go

```diff
@@ -58,57 +64,230 @@ func (r *Rest) initCache() error {
+	timestampRegex := `[+-]?\d{4}(-[01]\d(-[0-3]\d(T[0-2]\d:[0-5]\d:?([0-5]\d(\.\d+)?)?[+-][0-2]\d:[0-5]\d?)?)?)?`
+
+	regexTimeStamp := regexp.MustCompile(timestampRegex)
+	if match := regexTimeStamp.MatchString(value); match {
+		// example: 2020-12-02T18:36:19-08:00   ==>  1606962979
+		if timestamp, err = time.Parse(time.RFC3339, value); err != nil {
+			fmt.Printf("%v", err)
+			return 0
+		}
+		return float64(timestamp.Unix())
+	}
+	return 0
+}
+
+func (r *Rest) ParseRestCounters(counter *node.Node, prop *prop) {
+	var (
+		display, name, kind, metricType string
+	)
+
+	for _, c := range counter.GetAllChildContentS() {
 		if c != "" {
-			name, display, kind = util.ParseMetric(c)
+			name, display, kind, metricType = util.ParseMetric(c)
 			r.Logger.Debug().
 				Str("kind", kind).
 				Str("name", name).
 				Str("display", display).
 				Msg("Collected")
 
-			r.prop.counters[name] = display
+			prop.counters[name] = display
 			switch kind {
 			case "key":
-				r.prop.instanceLabels[name] = display
-				r.prop.instanceKeys = append(r.prop.instanceKeys, name)
+				prop.instanceLabels[name] = display
+				prop.instanceKeys = append(prop.instanceKeys, name)
 			case "label":
-				r.prop.instanceLabels[name] = display
+				prop.instanceLabels[name] = display
 			case "float":
-				m := metric{label: display, name: name}
-				r.prop.metrics = append(r.prop.metrics, m)
+				m := metric{label: display, name: name, metricType: metricType}
+				prop.metrics = append(prop.metrics, m)
 			}
 		}
 	}
 
-	// private end point do not support * as fields. We need to pass fields in endpoint
-	query := r.Params.GetChildS("query")
-	r.prop.apiType = "public"
-	if query != nil {
-		r.prop.apiType = checkQueryType(query.GetContentS())
-	}
-
-	if r.prop.apiType == "private" {
-		counterKey := make([]string, len(r.prop.counters))
+	if prop.apiType == "private" {
+		counterKey := make([]string, len(prop.counters))
 		i := 0
-		for k := range r.prop.counters {
+		for k := range prop.counters {
 			counterKey[i] = k
 			i++
 		}
-		r.prop.fields = counterKey
+		prop.fields = counterKey
 	}
 
-	if r.prop.apiType == "public" {
-		r.prop.fields = []string{"*"}
-		if c := r.Params.GetChildS("counters"); c != nil {
-			if x := c.GetChildS("hidden_fields"); x != nil {
-				r.prop.fields = append(r.prop.fields, x.GetAllChildContentS()...)
+	if prop.apiType == "public" {
+		prop.fields = []string{"*"}
+		if counter != nil {
+			if x := counter.GetChildS("hidden_fields"); x != nil {
+				prop.fields = append(prop.fields, x.GetAllChildContentS()...)
 			}
 		}
 	}
 
-	r.Logger.Info().Strs("extracted Instance Keys", r.prop.instanceKeys).Msg("")
-	r.Logger.Info().Int("count metrics", len(r.prop.metrics)).Int("count labels", len(r.prop.instanceLabels)).Msg("initialized metric cache")
+}
 
-	return nil
+func (r *Rest) HandleLabelsAndMetrics(instance *matrix.Instance, prop *prop, instanceData gjson.Result, data *matrix.Matrix, instanceKey string) uint64 {
```

Comment:
should be in Rest.go. Nothing to do with template parsing.

Reply:
done